### PR TITLE
rename src/fsharp/vs --> src/fsharp/service

### PR DESCRIPTION
**Pull request by dsyme**
Rename ``src/fsharp/vs`` since this directory is now the FCS language service consumed by many editing clients

### CODE REVIEW

**Comment by cartermp** [comment]
YES


On Fri, Dec 15, 2017, at 07:33, Don Syme wrote:
> Rename src/fsharp/vs since this directory is now the FCS language
> service consumed by many editing clients> 
> You can view, comment on, or merge this pull request online at:


>   https://github.com/Microsoft/visualfsharp/pull/4119


> Commit Summary


>  * make tuples support Item* with warning
>  * add deactivated tests for struct tuple
>  * Merge branch 'master' of http://github.com/Microsoft/visualfsharp
>    into tup1
>  * proper return types for Rest, prototype ctors
>  * fix SRTP
>  * fix tests
>  * Merge branch 'master' of http://github.com/Microsoft/visualfsharp
>    into tup1
>  * rename src/fsharp/vs --> src/fsharp/service> File Changes


>  * *M* .gitignore[1] (6)
>  * *M* fcs/FSharp.Compiler.Service.netstandard/FSharp.Compiler.Servic-
>    e.netstandard.fsproj[2] (64)
>  * *M* fcs/FSharp.Compiler.Service/FSharp.Compiler.Service.fspr-
>    oj[3] (64)
>  * *M* fcs/docsrc/content/ja/untypedtree.fsx[4] (2)
>  * *M* fcs/docsrc/content/queue.fsx[5] (2)
>  * *M* src/buildfromsource/FSharp.Compiler.Private/FSComp.fs[6] (4)
>  * *M* src/buildfromsource/FSharp.Compiler.Private/FSComp.resx[7] (3)
>  * *M* src/buildfromsource/FSharp.Compiler.Private/FSharp.Compiler.Pr-
>    ivate.fsproj[8] (64)
>  * *M* src/fsharp/AccessibilityLogic.fs[9] (4)
>  * *M* src/fsharp/AttributeChecking.fs[10] (10)
>  * *M* src/fsharp/ConstraintSolver.fs[11] (9)
>  * *M* src/fsharp/FSComp.txt[12] (1)
>  * *M* src/fsharp/FSharp.Compiler.Private/FSharp.Compiler.Private.fsp-
>    roj[13] (64)
>  * *M* src/fsharp/InfoReader.fs[14] (179)
>  * *M* src/fsharp/MethodOverrides.fs[15] (2)
>  * *M* src/fsharp/NameResolution.fs[16] (14)
>  * *M* src/fsharp/TastOps.fs[17] (38)
>  * *M* src/fsharp/TastOps.fsi[18] (20)
>  * *M* src/fsharp/TcGlobals.fs[19] (35)
>  * *M* src/fsharp/TypeChecker.fs[20] (13)
>  * *M* src/fsharp/fsi/fsi.fs[21] (2)
>  * *M* src/fsharp/infos.fs[22] (101)
>  * *R* src/fsharp/service/ExternalSymbol.fs[23] (0)
>  * *R* src/fsharp/service/ExternalSymbol.fsi[24] (0)
>  * *R* src/fsharp/service/IncrementalBuild.fs[25] (0)
>  * *R* src/fsharp/service/IncrementalBuild.fsi[26] (0)
>  * *R* src/fsharp/service/QuickParse.fs[27] (0)
>  * *R* src/fsharp/service/QuickParse.fsi[28] (0)
>  * *R* src/fsharp/service/Reactor.fs[29] (0)
>  * *R* src/fsharp/service/Reactor.fsi[30] (0)
>  * *R* src/fsharp/service/ServiceAnalysis.fs[31] (0)
>  * *R* src/fsharp/service/ServiceAnalysis.fsi[32] (0)
>  * *R* src/fsharp/service/ServiceAssemblyContent.fs[33] (0)
>  * *R* src/fsharp/service/ServiceAssemblyContent.fsi[34] (0)
>  * *R* src/fsharp/service/ServiceConstants.fs[35] (0)
>  * *R* src/fsharp/service/ServiceDeclarationLists.fs[36] (0)
>  * *R* src/fsharp/service/ServiceDeclarationLists.fsi[37] (0)
>  * *R* src/fsharp/service/ServiceInterfaceStubGenerator.fs[38] (0)
>  * *R* src/fsharp/service/ServiceInterfaceStubGenerator.fsi[39] (0)
>  * *R* src/fsharp/service/ServiceLexing.fs[40] (0)
>  * *R* src/fsharp/service/ServiceLexing.fsi[41] (0)
>  * *R* src/fsharp/service/ServiceNavigation.fs[42] (0)
>  * *R* src/fsharp/service/ServiceNavigation.fsi[43] (0)
>  * *R* src/fsharp/service/ServiceParamInfoLocations.fs[44] (0)
>  * *R* src/fsharp/service/ServiceParamInfoLocations.fsi[45] (0)
>  * *R* src/fsharp/service/ServiceParseTreeWalk.fs[46] (0)
>  * *R* src/fsharp/service/ServiceStructure.fs[47] (0)
>  * *R* src/fsharp/service/ServiceStructure.fsi[48] (0)
>  * *R* src/fsharp/service/ServiceUntypedParse.fs[49] (0)
>  * *R* src/fsharp/service/ServiceUntypedParse.fsi[50] (0)
>  * *R* src/fsharp/service/ServiceXmlDocParser.fs[51] (0)
>  * *R* src/fsharp/service/ServiceXmlDocParser.fsi[52] (0)
>  * *R* src/fsharp/service/service.fs[53] (4)
>  * *R* src/fsharp/service/service.fsi[54] (0)
>  * *M* src/fsharp/symbols/SymbolHelpers.fs[55] (10)
>  * *M* src/fsharp/symbols/Symbols.fs[56] (14)
>  * *M* src/fsharp/xlf/FSComp.txt.cs.xlf[57] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.de.xlf[58] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.en.xlf[59] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.es.xlf[60] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.fr.xlf[61] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.it.xlf[62] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.ja.xlf[63] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.ko.xlf[64] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.pl.xlf[65] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.pt-BR.xlf[66] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.ru.xlf[67] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.tr.xlf[68] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.zh-Hans.xlf[69] (5)
>  * *M* src/fsharp/xlf/FSComp.txt.zh-Hant.xlf[70] (5)
>  * *M* tests/fsharp/core/csext/test.fsx[71] (260)
>  * *M* tests/fsharp/typecheck/sigs/neg101.bsl[72] (74)
>  * *M* tests/fsharp/typecheck/sigs/neg101.fs[73] (41)
>  * *M* tests/fsharp/typecheck/sigs/pos28.fs[74] (2)
>  * *M* tests/fsharpqa/Source/Conformance/ObjectOrientedTypeDefinition-
>    s/TypeExtensions/basic/GenericExtensions.fs[75] (58)
>  * *M* tests/fsharpqa/Source/Conformance/ObjectOrientedTypeDefinition-
>    s/TypeExtensions/basic/GenericExtensionsCSLib.cs[76] (16)
>  * *M* vsintegration/Utils/LanguageServiceProfiling/Options.fs[77]
>    (36)
>  * *M* vsintegration/src/FSharp.LanguageService.Base/Source.cs[78] (2)> Patch Links:


>  * https://github.com/Microsoft/visualfsharp/pull/4119.patch
>  * https://github.com/Microsoft/visualfsharp/pull/4119.diff> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[79], or mute the
> thread[80].> 

Links:

   1. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-0
   2. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-1
   3. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-2
   4. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-3
   5. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-4
   6. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-5
   7. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-6
   8. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-7
   9. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-8
  10. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-9
  11. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-10
  12. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-11
  13. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-12
  14. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-13
  15. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-14
  16. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-15
  17. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-16
  18. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-17
  19. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-18
  20. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-19
  21. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-20
  22. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-21
  23. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-22
  24. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-23
  25. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-24
  26. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-25
  27. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-26
  28. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-27
  29. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-28
  30. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-29
  31. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-30
  32. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-31
  33. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-32
  34. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-33
  35. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-34
  36. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-35
  37. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-36
  38. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-37
  39. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-38
  40. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-39
  41. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-40
  42. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-41
  43. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-42
  44. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-43
  45. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-44
  46. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-45
  47. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-46
  48. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-47
  49. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-48
  50. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-49
  51. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-50
  52. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-51
  53. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-52
  54. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-53
  55. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-54
  56. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-55
  57. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-56
  58. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-57
  59. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-58
  60. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-59
  61. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-60
  62. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-61
  63. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-62
  64. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-63
  65. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-64
  66. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-65
  67. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-66
  68. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-67
  69. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-68
  70. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-69
  71. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-70
  72. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-71
  73. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-72
  74. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-73
  75. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-74
  76. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-75
  77. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-76
  78. https://github.com/Microsoft/visualfsharp/pull/4119/files#diff-77
  79. https://github.com/Microsoft/visualfsharp/pull/4119

**Comment by realvictorprm** [comment]
Finally :-)

Am 15.12.2017 17:56 schrieb "Don Syme" <notifications@github.com>:

File: .gitignore [hunk 1]
```diff
@@ -114,7 +114,7 @@ tests/fsharp/typecheck/sigs/neg94-pre.dll
 times
 /tests/fsharpqa/testenv/bin/System.ValueTuple.dll
 source_link.json
-.vs/
+.service/
```
Comment (dhwed) [inline .gitignore]: This is an accident; `.vs/` needs to be added back here

Comment (dsyme) [reply .gitignore]: Good catch! :)

**Pull request merged**